pkg/ccl/changefeedccl/schemafeed/schema_feed.go

```diff
@@ -644,10 +644,15 @@ func (tf *schemaFeed) adjustTimestamps(startTS, endTS hlc.Timestamp, validateErr
 		}
 		return validateErr
 	}
-
-	if frontier := tf.mu.ts.frontier; frontier.Less(startTS) {
+	frontier := tf.mu.ts.frontier
+	if frontier.Less(startTS) {
 		return errors.Errorf(`gap between %s and %s`, frontier, startTS)
 	}
+	// If the current frontier is greater than the endTS,
+	// then we do not need to advance the frontier.
```

Comment:
Could you expand this comment to explain why this could happen? From a quick glance it seems to be from the `pauseOrResumePolling` logic when the table is `schema_locked`. Does that match up with your understanding?

Reply:
I purposefully left out how not having this could lead to errors from the comment, in order to keep the function isolated from caller's and callee's behavior. I added a note that not checking could lead to errors in `advanceFrontier`. I added more on the hypothesis about what specific functions are racing in the commit message instead.